nncf/onnx/graph/node_utils.py

```diff
@@ -209,3 +213,20 @@ def get_quantized_tensor_shape(
     if target_point.is_weight_target_point():
         return node.layer_attributes.weight_attrs[target_point.port_id]["shape"]
     return _get_activation_tensor_shape(nncf_graph, node, target_point)
+
+
+def calculate_gemm_channel_axis(node: NNCFNode, port_id: int) -> int:
+    """
+    Calculates Gemm channel axis based on the port and node attributes.
+
+    :param node: NNCFNode instance.
+    :param port_id: Port ID.
+    :return: Channel axis number.
+    """
+    # Gemm metatype supports only 2D inputs according to the documentation -
+    # https://onnx.ai/onnx/operators/onnx__Gemm.html
+    gemm_shape = [0, 1]
+    trans_attr = "transA" if not port_id else "transB"
+    transpose = node.layer_attributes.node_attrs[trans_attr]
+    channel_axis = -1 - port_id if transpose else -2 + port_id
```

Comment:
I agree that this approach is now aligned with OpenVINO, but why did you choose the logic of axis processing of OpenVINO instead of ONNX? For me, it seems that ONNX supports all input shapes, while OV doesn't. But maybe I am wrong...